apps/website/src/pages/show-and-tell/my-posts/[postId]/preview.tsx

```diff
@@ -72,13 +75,31 @@ const PreviewShowAndTellPage: NextPage = () => {
 
         {session?.status === "authenticated" && (
           <>
-            {getMyPost.isLoading && <p>Loading...</p>}
-            {getMyPost.isError && (
-              <MessageBox variant="failure">
-                {getMyPost.error.message}
-              </MessageBox>
+            {getMyPost.isLoading || getPostsFromANewLocation.isLoading ? (
+              <p>Loading...</p>
+            ) : getMyPost.isError || getPostsFromANewLocation.isError ? (
+              <>
+                {getMyPost.isError && (
+                  <MessageBox variant="failure">
+                    {getMyPost.error.message}
+                  </MessageBox>
+                )}
+                {getPostsFromANewLocation.isError && (
+                  <MessageBox variant="failure">
+                    {getPostsFromANewLocation.error.message}
+                  </MessageBox>
+                )}
+              </>
+            ) : (
+              getMyPost.data && (
+                <ShowAndTellEntry
+                  entry={getMyPost.data}
+                  newLocation={getPostsFromANewLocation.data.has(
+                    getMyPost.data.id,
+                  )}
+                />
+              )
             )}
```

Comment:
I'm not sure I'm a fan of the refactor into a chained ternary here -- I think it was cleaner as a set of standalone conditionals

Reply:
I believe the query states (e.g. isLoading) are mutually exclusive. I wanted to ensure the user was not shown an error and an incomplete entry to avoid confusion.

Reply:
Yeh, I do believe they're mutually exclusive, and agree that the nested ternary helps enforce that. That being said, I do think the individual blocks is a lot cleaner to read than a nested ternary, and given they should be mutually exclusive from the hook already, I'd prefer sticking to the old style, please?

Reply:
What do you mean by mutually exclusive from the hook?

I could create a couple variables that track the states of both queries and revert the nested ternary.

Reply:
As in, if a given hook is in the loading state, I don't _think_ it'll also be in the error state.

If one of the hooks is loading and the other is erroring, then I think we'd want to show the user both messages, not a cascading ternary where we'd just show a loading message and no error (or visa-versa).

I don't think any additional variables would be needed, your first block would just check if either is loading and show a loading message, the second checking if either are erroring and show an error message, and the third if the data is present to actually render the entry?

Reply:
That's my assumption as well.

I thought it might be confusing to show a loading state and display an error, but maybe I'm overthinking it.

Reply:
I think it's probably fine, and hopefully very much an edge-case we'll never run into, so I'd prefer to lean into cleaner/easier-to-read code with the isolated blocks rather than a nested tenary :nod:

Reply:
Is f37dff3 what you had in mind?

Reply:
Yes, thank you!